cleanlab/segmentation/filter.py

```diff
@@ -163,34 +151,31 @@ def get_images_to_load(pre_pred_probs, i, n, batch_size):
 
         pbar = tqdm(desc="number of examples processed for estimating thresholds", total=n)
 
-    i = 0
-    while i < n:
-        images_to_load = get_images_to_load(pre_pred_probs, i, n, batch_size)
-        end_index = i + images_to_load
+    # Precompute the size of each image in the batch
+    image_size = np.prod(pre_pred_probs.shape[1:])
+    images_per_batch = max(batch_size // image_size + 1, 0)
```

Comment:
just curious: why is max(x, 0) needed here?  isnt the x never negative?

Reply:
also you are sure the lower-bound here should be 0 and not 1?

Reply:
Hmm... I had a notion that 'batch_size' could possibly take negative values, but I can't recall the exact reasoning now (my guess is that something like `batch_size = -1` was breaking tests, but is now disallowed). Switching to 

```python
images_per_batch = np.max(batch_size // image_size, 1)
```